library/modules/Gui.cpp

```diff
@@ -650,7 +649,7 @@ bool Gui::item_details_hotkey(df::viewscreen *top)
 
 static bool has_cursor()
 {
-    return df::global::cursor && df::global::cursor->x != -30000;
+    return Gui::getCursorPos().isValid();
```

Comment:
I think a better option (for now, anyway) would be to check for UI modes where the cursor is known to be valid, since we can't determine validity solely from looking at the cursor.

Reply:
I'm not sure if that's the right strategy. We're already talking about having the cursor set by clicking or cursor follows mouse. That's explicitly so we can have valid cursors outside of the known good modes.

Reply:
That is fair, so it would need to check for that somehow. This comment was really conditional on https://github.com/DFHack/dfhack/pull/2657#discussion_r1080776011, though, so you can disregard this one.